### PR TITLE
Added manual CSR approval flow

### LINKED ISSUE #557: Allow manual CSR approval
**User Story**

If the user opts for manual approval, the `ByoAdmission Controller` should not run. The Platform Operator can opt for manual approval by overriding env variables when installing the BYOH provider.

**Solution**
Flag out the ByoAdmission controller behind an environment variable.

### PR DESCRIPTION
**Pull request by sachinkumarsingh092**
Signed-off-by: Sachin Kumar Singh <sachins2@vmware.com>

**What this PR does / why we need it**:
It should be possible to approve CSR manually. For that, we will disable the ByoAdmission controller by applying it behind an environment variable. Now user can apply the variable `MANUAL_CSR_APPROVAL=true` to manually approve the CSR. By default, the controller will approve the CSRs.

File changes:
- **main.go**: Put the ByoAdmission controller setup behind the `MANUAL_CSR_APPROVAL` environment variable.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #557 

**Additional information**
I've disabled the cyclometric complexity form `main()` due to lots of inevitable `if` conditions that might even increase in the future and thus increase the complexity further. 
Another way is to refactor the controller and webhook setup out of the main.

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -125,18 +130,21 @@ func main() {
...
-	if err = (&byohcontrollers.ByoAdmissionReconciler{
-		ClientSet: clientset.NewForConfigOrDie(ctrl.GetConfigOrDie()),
-	}).SetupWithManager(mgr); err != nil {
-		setupLog.Error(err, "unable to create controller", "controller", "ByoAdmission")
-		os.Exit(1)
-	}
 
+	// Set 'MANUAL_CSR_APPROVAL=true' to disable ByoAdmission controller. Now CSRs should be approved manually.
+	manualCSRApproval, ok := os.LookupEnv("MANUAL_CSR_APPROVAL")
+	if !ok || manualCSRApproval != "true" {
```
Comment (mayur-tolexo) [inline main.go:136]: Nit:
```suggestion
	if  os.Getenv("MANUAL_CSR_APPROVAL") != "true" {
```

**Review by dharmjit — APPROVED** [review]
LGTM, we need to update docs for this environment variable and bootstrap kubeconfig flow in general. Could we add some tests for this behavior?

**Comment by sachinkumarsingh092** [comment]
Should I write an e2e test for this? I don’t see any tests for the manager otherwise.

**Review by anusha94 — APPROVED** [review]
lgtm

**Pull request merged**